### PR TITLE
Upgrade astroid on 2.17.x branch

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
Backport of #9071.

### CODE REVIEW

**Comment by jacobtylerwalls** [comment]
@mbyrnepr2 Do you mind taking over and fixing the 3.7 test? 🙏

**Comment by mbyrnepr2** [comment]
Happy to do that @jacobtylerwalls and thanks for doing all this work related to issues of mine recently!
Is it necessary to backport tests like this? Our main branch CI doesn't run tests for Python 3.7 anymore since it is end of life so does it make sense to update the tests to handle that?

**Comment by jacobtylerwalls** [comment]
Good point, no, not necessarily, unless the test reveals a real incompatibility with the stuff we're about to backport. Sounds like no (i.e. just an issue with the test having slightly different output?)

**Comment by Pierre-Sassoulas** [comment]
I think it's ok to more or less silently "drop support for python 3.7" (i.e. simply removing the tests, not actually doing what we did on main) in the latest 2.17.x. Especially if we still support 2.17.x it at the same time than 3.0.x maintenance branch following 3.0.0 release.

**Comment by jacobtylerwalls** [comment]
Alright, let's just merge the astroid upgrade, and do a pylint patch release.

> Especially if we still support 2.17.x it at the same time than 3.0.x maintenance branch following 3.0.0 release.

-1 on merging anything else to the 2.17.x maintenance branch ever again, honestly.

File: doc/whatsnew/fragments/9069.false_positive [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Fix a regression in pylint 2.17.6 / astroid 2.15.7 causing various
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/fragments/9069.false_positive:1]: Isn't the message going to be in the backport of the fix ?

**Review by Pierre-Sassoulas — APPROVED** [review]
Nevermind :)

**Pull request merged**